codec-http/src/main/java/io/netty/handler/codec/http/HttpStatusClass.java

```diff
@@ -74,6 +74,29 @@ public static HttpStatusClass valueOf(int code) {
         return UNKNOWN;
     }
 
+    /**
+     * Returns the class of the specified HTTP status code.
+     * @param code Just the numeric portion of the http status code.
+     */
+    public static HttpStatusClass valueOf(CharSequence code) {
+        if (code != null && code.length() == 3) {
+            char c0 = code.charAt(0);
+            char c1 = code.charAt(1);
+            char c2 = code.charAt(2);
+            return isDigit(c0) && isDigit(c1) && isDigit(c2) ?
+                    valueOf(digit(c0) * 100 + digit(c1) * 10 + digit(c2)) : UNKNOWN;
```

Comment:
I was trying to balance how much was determined in this method vs delegating to `valueOf(int)`. However since both are in the scope of this `Enum` I guess I can be more aggressive here and duplication of logic won't be too bad.